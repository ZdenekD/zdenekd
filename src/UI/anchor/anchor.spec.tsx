import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../__test__/utils/helpers';
import Anchor from '.';

describe('Anchor', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Anchor href="#">content</Anchor>);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-anchor');

        expect(component.exists()).toBe(true);
    });

    it('has rel attribute if target is set', () => {
        const anchor = shallow(<Anchor href="/" target="_blank">content</Anchor>);
        const component = findComponent(anchor, 'component-anchor');

        expect(component.html().includes('rel')).toBe(true);
    });
});
