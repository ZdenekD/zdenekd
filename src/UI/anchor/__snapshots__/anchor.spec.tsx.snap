// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor match snapshot 1`] = `
<a href="#"
   class="anchor "
   data-test="component-anchor"
>
  content
</a>
`;
